velox/functions/prestosql/aggregates/tests/AggregationTestBase.cpp

```diff
@@ -32,4 +35,135 @@ std::vector<RowVectorPtr> AggregationTestBase::makeVectors(
   return vectors;
 }
 
+void AggregationTestBase::testAggregations(
+    const std::vector<RowVectorPtr>& data,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::string& duckDbSql) {
+  testAggregations(data, groupingKeys, aggregates, {}, duckDbSql);
+}
+
+void AggregationTestBase::testAggregations(
+    const std::vector<RowVectorPtr>& data,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& postAggregationProjections,
+    const std::string& duckDbSql) {
+  testAggregations(
+      [&](PlanBuilder& builder) { builder.values(data); },
+      groupingKeys,
+      aggregates,
+      postAggregationProjections,
+      duckDbSql);
+}
+
+void AggregationTestBase::testAggregations(
+    std::function<void(PlanBuilder&)> makeSource,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::string& duckDbSql) {
+  testAggregations(makeSource, groupingKeys, aggregates, {}, duckDbSql);
+}
+
+void AggregationTestBase::testAggregations(
+    std::function<void(PlanBuilder&)> makeSource,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& postAggregationProjections,
+    const std::string& duckDbSql) {
+  // Run partial + final.
+  {
+    PlanBuilder builder;
+    makeSource(builder);
+    builder.partialAggregation(groupingKeys, aggregates).finalAggregation();
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+    assertQuery(builder.planNode(), duckDbSql);
+  }
+
+  // Run single.
+  {
+    PlanBuilder builder;
+    makeSource(builder);
+    builder.singleAggregation(groupingKeys, aggregates);
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+    assertQuery(builder.planNode(), duckDbSql);
+  }
+
+  // Run partial + intermediate + final.
+  {
+    PlanBuilder builder;
+    makeSource(builder);
+    builder.partialAggregation(groupingKeys, aggregates)
+        .intermediateAggregation()
+        .finalAggregation();
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+    assertQuery(builder.planNode(), duckDbSql);
+  }
+
+  // Run partial + local exchange + final.
+  if (!groupingKeys.empty()) {
```

Comment:
Can't we use `localPartitionRoundRobin` here if the groupingKeys are empty like below.?

Reply:
If we use localPartitionRoundRobin here for global aggregation, we'll end up with final aggregation producing 2 rows instead of 1.

Reply:
Thanks!